clients/src/main/java/org/apache/kafka/clients/consumer/internals/CommitRequestManager.java

```diff
@@ -190,63 +179,90 @@ private static long findMinTime(final Collection<? extends RequestState> request
+            log.debug("Skipping auto-commit, remaining time {}", autocommit.timer.remainingMs());
             return CompletableFuture.completedFuture(null);
         }
 
-        CompletableFuture<Void> result = sendAutoCommit(offsets);
+        CompletableFuture<Void> result = addOffsetCommitRequest(offsets, timer).whenComplete(autoCommitCallback(offsets));
         autocommit.resetTimer();
         autocommit.setInflightCommitStatus(true);
         return result;
     }
 
     /**
      * If auto-commit is enabled, this will generate a commit offsets request for all assigned
-     * partitions and their current positions.
+     * partitions and their current positions. Note on auto-commit timers: this will reset the
+     * auto-commit timer to the interval before issuing the async commit, and when the async commit
+     * completes, it will reset the auto-commit timer with the exponential backoff if the request
+     * failed with a retriable error.
      *
      * @return Future that will complete when a response is received for the request, or a
      * completed future if no request is generated.
      */
-    public CompletableFuture<Void> maybeAutoCommitAllConsumed() {
-        return maybeAutoCommit(subscriptions.allConsumed());
-    }
+    public CompletableFuture<Void> maybeAutoCommitAllConsumedAsync() {
+        if (!autoCommitEnabled()) {
+            // Early return to ensure that no action/logging is performed.
+            return CompletableFuture.completedFuture(null);
+        }
+        Map<TopicPartition, OffsetAndMetadata> offsets = subscriptions.allConsumed();
+        CompletableFuture<Void> result = maybeAutoCommit(offsets, Optional.empty(), true);
+        result.whenComplete((__, error) -> {
+            if (error != null) {
+                if (error instanceof RetriableCommitFailedException) {
+                    log.debug("Asynchronous auto-commit of offsets {} failed due to retriable error.", offsets, error);
+                    resetAutoCommitTimer(retryBackoffMs);
+                } else {
+                    log.warn("Asynchronous auto-commit of offsets {} failed: {}", offsets, error.getMessage());
+                }
+            } else {
+                log.debug("Completed asynchronous auto-commit of offsets {}", offsets);
+            }
+        });
 
-    boolean canAutoCommit() {
-        return autoCommitState.isPresent() && !subscriptions.allConsumed().isEmpty();
+        return result;
     }
 
     /**
-     * Updates the member ID and epoch upon receiving ConsumerGroupHeartbeatResponse.
+     * Commit consumed offsets it auto-commit is enabled. Retry while the timer is not expired,
+     * until the request succeeds or fails with a fatal error.
      */
-    void updateMemberInformation(String memberId, int memberEpoch) {
-        groupState.generation = new GroupState.Generation(memberEpoch, memberId, null);
+    public CompletableFuture<Void> autoCommitAllConsumedNow(Optional<Timer> timer) {
```

Comment:
I suppose to be consistent, this shoukld be called `maybeAutoCommitAllConsumedNow`